src/utils/model_utils.py

```diff
@@ -83,7 +83,11 @@ def train(
                 model, optim, dloader, loss_fn, device, test_loader=None, **kwargs
             )
             return mean_loss, acc
-
+        elif kwargs.get("malicious_type", None) == "backdoor_attack" or kwargs.get("malicious_type", None) == "gradient_attack":
```

Comment:
Instead of passing it in the train arguments, we should make it a member of the class by passing it when creating `model_utils`
That way different algos don't have to change their train function explicitly.